src/test/libcephfs/test.cc

```diff
@@ -2640,3 +2640,25 @@ TEST(LibCephFS, LookupVino) {
 
   ceph_shutdown(cmount);
 }
+
+TEST(LibCephFS, LookupMdsPrivateInos) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, NULL), 0);
+
+  Inode *inode;
+  for (int ino = 0; ino < MDS_INO_SYSTEM_BASE; ino++) {
+    if (MDS_IS_PRIVATE_INO(ino)) {
+      ASSERT_EQ(-ESTALE, ceph_ll_lookup_inode(cmount, ino, &inode));
+    } else if (ino == CEPH_INO_ROOT) {
+      ASSERT_EQ(0, ceph_ll_lookup_inode(cmount, ino, &inode));
+    } else if (ino == CEPH_INO_LOST_AND_FOUND) {
+      int r = ceph_ll_lookup_inode(cmount, ino, &inode);
+      ASSERT_TRUE(r == -ESTALE || r == 0);
```

Comment:
Shouldn't this just be asserting that r == 0 here? If it returns ESTALE on that one, it's a bug, no?

Reply:
With a fresh fs cluster the `lost+found` dir inode will be absent and only after running the recovery tool will it be created.

If absent the MDS will return `-ESTALE`.

Reply:
Got it. Thanks.